docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+
+### `Shutdown()`
+
+Shuts down the exporter. Called when SDK is shut down. This is an opportunity for exporter to do any cleanup required.
+
+`Shutdown` should be called only once for each `Exporter` instance. After the call to `Shutdown` subsequent calls to `Export` are not allowed and should return a `Failure` result.
+
+`Shutdown` should not block indefinitely (e.g. if it attempts to flush the data and the destination is unavailable). Language library authors can decide if they want to make the shutdown timeout configurable.
+
+In the StdoutExporter there is no cleanup to be done, so there is no need to use the timeout within the `Shutdown` function as it will never be blocking. 
+
+```
+class StdoutSpanExporter final : public sdktrace::SpanExporter
+{
+
+private: 
+    bool isShutdown = false;
+    
+public:
+    /* 
+        This function should never be called concurrently.
+    */ 
+    sdktrace::ExportResult Export(
+        const nostd::span<std::unique_ptr<sdktrace::Recordable>> &spans) noexcept
+    {
+    
+        Time startTime = time.now();
+    
+        if(isShutdown)
+        {
+            return sdktrace::ExportResult::kFailure;
+        }
+    
+        for (auto &recordable : spans)
+        {
+            auto span = std::unique_ptr<sdktrace::SpanData>(
+                static_cast<sdktrace::SpanData *>(recordable.release()));
+        
+            if (span != nullptr)
+            {
+                char trace_id[32]       = {0};
+                char span_id[16]        = {0};
+                char parent_span_id[16] = {0};
+        
+                span->GetTraceId().ToLowerBase16(trace_id);
+                span->GetSpanId().ToLowerBase16(span_id);
+                span->GetParentSpanId().ToLowerBase16(parent_span_id);
+        
+                std::cout << "{"
+                        << "\n  name          : " << span->GetName()
+                        << "\n  trace_id      : " << std::string(trace_id, 32)
+                        << "\n  span_id       : " << std::string(span_id, 16)
+                        << "\n  parent_span_id: " << std::string(parent_span_id, 16)
+                        << "\n  start         : " << span->GetStartTime().time_since_epoch().count()
+                        << "\n  duration      : " << span->GetDuration().count() 
+                        << "\n  description   : " << span->GetDescription() 
+                        << "\n  status        : " << span->GetStatus()
+                        << "\n  attributes    : " << span->GetAttributes() << "\n}"
+                        << "\n";
+            }
+            
+            if(time.now() - startTime > 30s) {
+                return sdktrace::ExportResut::kFailure;
```

Comment:
I think we can skip the timeout feature, because
* we can rely on the fact that printing to `stdout` is reasonably performant and doesn't block, and
* this exporter will not be used in production code, but rather in explorations or examples (so there are benefits to keeping the implementation as simple as possible).

I think it's safe enough to rely on `std::cout` to not block indefinitely.

Reply:
Talked to Reiley today and totally agree that this is unnecessary. Will take it out of the document and implementation.

Reply:
FYI, in OpenTelemetry .NET the console exporter [doesn't have timeout](https://github.com/open-telemetry/opentelemetry-dotnet/blob/4abc57d02e213d958b4ed5900f8921006f9df23e/src/OpenTelemetry.Exporter.Console/ConsoleExporter.cs#L114), and [doesn't aim to have synchronization on stdout writes](https://github.com/open-telemetry/opentelemetry-dotnet/pull/783#discussion_r451320435).